app/services/age_calculator.rb

```diff
@@ -1,6 +1,6 @@
 class AgeCalculator
-  def self.call(date_of_birth, date)
-    age = date.year - date_of_birth.year
-    date_of_birth > date.years_ago(age) ? age - 1 : age
+  def self.call(date_of_birth, as_of)
+    age = as_of.year - date_of_birth.year
+    date_of_birth > as_of.years_ago(age) ? age - 1 : age
   end
 end
```

Comment:
I appreciate we use the so called "service object" pattern throughout the application, but I wish we could call it like this instead

```ruby
Age.calculate(date_of_birth, as_of)
```

Reply:
Also, would it be clearer if the method was more explicit?

```ruby
class Age
  def self.calculate(date_of_birth, as_of)
    years_since_birth = as_of.year - date_of_birth.year
    next_birthday = date_of_birth.years_since(years_since_birth)
    
    if as_of.before?(next_birthday)
      years_since_birth - 1
    else
      years_since_birth
    end
  end
end
```

Reply:
I was more interested in refactoring the spec myself and limited myself to only the minor change in the method/class itself.

Feel free to PR one for discussion re: services versus "utility" classes (?!) - i would probably move it out of the services directory in this situation to lib or some such ?!

Reply:
There is also a gem i found btw, not that i would advocate its use, but interesting nonetheless
https://github.com/kenn/age_calculator